python/lsst/pipe/analysis/fakesAnalysis.py

```diff
@@ -0,0 +1,615 @@
+
+    catalog[raCol + "_deg"] = np.rad2deg(catalog[raCol].values)
+    catalog[decCol + "_deg"] = np.rad2deg(catalog[decCol].values)
+
+    return catalog
+
+
+def matchCatalogs(catalog1, raCol1, decCol1, catalog2, raCol2, decCol2, matchRadius=0.1/3600.0):
+    """Match two dataframes together by RA and Dec.
+
+    Parameters
+    ----------
+    catalog1 : `pandas.core.frame.DataFrame`
+        The catalog to be matched to catalog2.
+    catalog2 : `pandas.core.frame.DataFrame`
+        The catalog that catalo1 is matched to.
+    raCol1 : `string`
+        The column name for the RA column in catalog1.
+    decCol1 : `string`
+        The column name for the Dec. column in catalog1.
+    raCol2 : `string`
+        The column name for the RA column in catalog2.
+    decCol2 : `string`
+        The column name for the Dec. column in catalog2.
+    matchRadius : `float`
+        Radius within which to match the nearest source, in degrees, defaults to 0.1 arcseconds (0.1/3600.0).
+
+    Returns
+    -------
+    catalog1Matched : `pandas.core.frame.DataFrame`
+        Catalog with only the rows that had a match in catalog2.
+    catalog2Matched : `pandas.core.frame.DataFrame`
+        Catalog with only the rows that matched catalog1.
+
+    Notes
+    -----
+    matchRadius is in degrees, the catalogs need to be in degrees rather than radians. Returns two shortened
+    catalogs, with the matched rows in the same order and objects without a match removed. Matches the first
+    catalog to the second, multiple objects from the first catalog can match the same object in the second.
+    """
+
+    dists, inds = match_lists(catalog1[raCol1], catalog1[decCol1], catalog2[raCol2], catalog2[decCol2],
+                              matchRadius)
+
+    ids = (inds != len(catalog2))
+    matchedInds = inds[ids]
+    catalog1Matched = catalog1[ids]
+    catalog2Matched = catalog2.iloc[matchedInds].copy()
+
+    return catalog1Matched, catalog2Matched
+
+
+def addNearestNeighbor(catalog, raCol, decCol):
+    """Add the distance to the nearest neighbour in the catalog.
+
+    Parameters
+    ----------
+    catalog : `pandas.core.frame.DataFrame`
+        Catalog to add the distance to the nearest neighbour to.
+    raCol : `string`
+        Column name for the RA column to be used.
+    decCol : `string`
+        Column name for the Declination column to be used.
+
+    Returns
+    -------
+    catalog : `pandas.core.frame.DataFrame`
+        Catalog with a column 'nearestNeighbor' containing the distance to the neareast neighbour.
+
+    Notes
+    -----
+    Distance added is in degrees.
+    """
+
+    dists, inds = match_lists(catalog[raCol], catalog[decCol], catalog[raCol], catalog[decCol],
+                              20.0 / 3600, numNei=2)
+
+    distNN = np.array([d[1] for d in dists])
+    catalog["nearestNeighbor"] = distNN
+
+    return catalog
+
+
+def sigmaMAD(xs, med):
+    """Calculate the sigma given by the median absolute deviation.
```

Comment:
I've got good/bad news for you 🙃 https://docs.astropy.org/en/stable/api/astropy.stats.median_absolute_deviation.html